openapiart/bundler.py

```diff
@@ -635,6 +663,37 @@ def _required_restriction(self, schema_name, value):
                                 )
                             )
 
+    def _validate_x_field_pattern(self):
+        for xpattern_path in self._get_parser("$..x-field-pattern").find(
+            self._content
+        ):
+            xpattern = xpattern_path.value
+
+            if "format" not in xpattern:
+                self._errors.append(
+                    "{} is not valid, format is mandatory for x-field-pattern".format(
+                        str(xpattern_path.full_path)
+                    )
+                )
+            elif xpattern["format"] == "integer" and "length" not in xpattern:
+                self._errors.append(
+                    "{} property using x-field-pattern with format integer must contain length property".format(
+                        str(xpattern_path.full_path)
+                    )
+                )
+            valid_formats = ["integer", "ipv4", "ipv6", "mac", "checksum"]
+            if xpattern["format"] not in valid_formats:
+                self._errors.append(
+                    "%s has unspported format %s , valid formats are %s"
+                    % (
+                        str(xpattern_path.full_path),
+                        xpattern["format"],
+                        str(valid_formats),
+                    )
+                )
+        if len(self._errors) > 0:
+            self._validate_errors()
+
     def _resolve_x_pattern(self, pattern_extension):
```

Comment:
can we replace all variables called `format` with `fmt` ? it's a keyword in python

Reply:
done